### DISCUSSION TITLE
Printing in pendulum mode.

### DISCUSSION THREAD

**Discussion by Gudzonskiy** [OrcaSlicer/OrcaSlicer#5132] [Ideas]
Hello! There is a big request in future releases of the Orca slicer program to add the function of printing parts out of order, the so-called pendulum mode. The essence is this, for example, there are three models “A” “B” and “C” and so that the order of their printing is A-B-C, then raise one layer C-B-A, raise A-B-C-C-B-A and so on. 

P.S. Of all the slicers, I found such a function only in Ideamaker!

**Comment by vgdh** [comment]
It make sense to implement. 
Orca uses that approach when prints with different filament it helps to increase the speed of printing and reduce the number of filament change.
May be @Noisyfox of @SoftFever have some thoughts on that topic.

**Comment by Noisyfox** [comment]
I'm concerned about this because this could lead to uneven cooling.

**Comment by SoftFever** [comment]
I thought about this and abandoned this idea eventually because of these two reasons:
1. Cooling issue: Considering such a common case, the last thing we want to do is print the benchy chimney two layers in a row.
<img width="557" alt="Screenshot 2024-04-25 at 8 01 31 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/103989404/da2957ab-dc12-49e4-b9a8-42b720f843e8">

2. You don't really save much time as nowadays printers are capable of traveling at 300-600mm/s.

Also, uneven cooling as Noisyfox mentioned will also create visual artifacts

**Comment by Gudzonskiy** [comment]
I don't understand how one benchy ship is printed on such printers?

**Reply by igiannakas** [reply]
When you print a single object the whole layer is slowed down to meet layer time targets. 

However when printing two objects and the second one is large enough to exceed layer time limits, the smaller object would have significant amount of heat dumped into it because the layers would print consecutively without any slowdown.

to work around this you could apply layer time restrictions to the smaller object independently of the larger one, but then you’ve just slowed down the print, in which case might as well travel and print the larger object first ;)

**Comment by vgdh** [comment]
https://github.com/SoftFever/OrcaSlicer/issues/6651
